### PR TITLE
Test form

### PR DESCRIPTION
**Pull request by Allislove**

### CODE REVIEW

File: src/components/loader.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { InfinitySpin } from  'react-loader-spinner'
+
+
+
+export const Loader = () => {
```
Comment (Israel-Laguan) [inline src/components/loader.tsx]: Remove extra lines

File: src/components/questions.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  const [success, setSuccess] = useState(false)
+  // Opciones seleccionadas por el usuario
+  const [selectedOptions, setSelectedOptions] = useState({})
+  // Identificador de la pregunta actual
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0)
+
+  const handleGetJSON = async (event: any) => {
+    const file = event.target.files[0]
+    if (file) {
+      // console.log("File updated", file.name);
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove this line

File: src/components/questions.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      setLoading(true)
+      console.log(loading, '*** 01 Loading ***')
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/components/questions.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      const reader = new FileReader()
+      reader.onload = (event: any) => {
+        try {
+          const jsonData = JSON.parse(event.target.result)
+          // console.log(jsonData, "JSON DATA ***");
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/components/questions.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+          //console.log("Hello World!");
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/components/questions.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+          if (jsonData) {
+            let promise = new Promise((resolve, reject) => {
+              // Ejecutar primero la funcion del spinner
+              notify('Cargado de manera satisfactoria')
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Please use English

File: src/components/questions.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+            // Enviamos las preguntas al estado
+            setQuestions(jsonData.questions)
+            setSuccess(true) // JSON cargado de manera satisfactoria
+          }
+        } catch (error) {
+          notifyError('Archivo JSON Invalido')
+          setSuccess(false)
+          setError('JSON inválido')
+          // setLoading(false);
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove if not needed

File: src/components/questions.tsx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+      loading ? 'Loading...' : 'Get a JSON'
+    }
+    console.log(`${loading ? 'Loading...' : 'Get a JSON'};`)
+    setLoading(true)
+  }
+
+  useEffect(() => {
+    // Vamos guardando cada seleccion del input radio en el localStorage
+    let data = localStorage.setItem('selectedOptions', JSON.stringify(selectedOptions))
+    // console.log(data);
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/components/questions.tsx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+  const notify = (message: string) => {
+    toast.success(message, {
+      position: 'top-center',
+      autoClose: 3000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      draggable: true,
+    })
+    // console.log("JSON cargado de manera satisfactoria");
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/components/questions.tsx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+  const notifyError = (message: string) => {
+    toast.error(message, {
+      position: 'top-center',
+      autoClose: 3000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      draggable: true,
+    })
+    // console.log("Error al cargar el JSON");
```
Comment (Israel-Laguan) [inline src/components/questions.tsx]: Remove

File: src/pages/exam.tsx [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import QuestionsForm from "@/components/questions";
+
+
+
+export default function Exam() {
+
+  return(
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: Remove extra lines

File: src/pages/exam.tsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+    <>
+      <QuestionsForm />
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: This component is not really needed

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]

File: src/pages/exam.tsx (hunk 12 above)
Comment (Allislove) [reply src/pages/exam.tsx]: Put all into the principal ?

File: src/components/questions.tsx (hunk 8 above)
Comment (Allislove) [reply src/components/questions.tsx]: All the useEffect? @Israel-Laguan

File: src/components/questions.tsx (hunk 5 above)
Comment (Allislove) [reply src/components/questions.tsx]: lines 28-29?

**Comment by Allislove** [comment]
@Israel-Laguan  Can you review again please!

File: src/pages/exam.tsx (hunk 12 above)
Comment (Israel-Laguan) [reply src/pages/exam.tsx]: In this case yes

**Comment by Israel-Laguan** [comment]
Linter says that this file needs revision
image
Please update with linter

**Comment by Allislove** [comment]
@Israel-Laguan Done, all checks have passed.

File: src/pages/exam.tsx [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+  // Identificador de la pregunta actual
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0)
+  const [isOptionSelected, setIsoptionSelected] = useState(false)
+  const fileRef = useRef()
+
+  const handleGetJSON = async (event: any) => {
+    const file = event.target.files[0]
+    if (file) {
+      setLoading(true)
+      console.log(loading, '*** 01 Loading ***')
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: Remove

File: src/pages/exam.tsx [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+      // Leemos el archivo
+      reader.readAsText(file)
+    }
+  }
+
+  const loadingInfo = () => {
+    {
+      loading ? 'Loading...' : 'Get a JSON'
+    }
+    console.log(`${loading ? 'Loading...' : 'Get a JSON'};`)
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: Remove

File: src/styles/globals.css [hunk 15]
```diff
@@ -40,68 +40,83 @@
...
-  text-decoration: none;
+/* form {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  margin: 0 auto;
+  width: 100%;
+  max-width: 500px;
+  padding: 0 20px;
```
Comment (Israel-Laguan) [inline src/styles/globals.css]: Remove the commented code if not used

File: src/pages/exam.tsx (hunk 13 above)
Comment (Allislove) [reply src/pages/exam.tsx]: @Israel-Laguan  the console.log ?

File: src/pages/exam.tsx (hunk 14 above)
Comment (Allislove) [reply src/pages/exam.tsx]: Console.log ?

Comment (Israel-Laguan) [reply src/pages/exam.tsx]: Yes, not needed

File: src/pages/exam.tsx (hunk 13 above)
Comment (Israel-Laguan) [reply src/pages/exam.tsx]: yes

**Comment by Allislove** [comment]
@Israel-Laguan  Done, can you review again plz

**Review by Israel-Laguan — APPROVED** [review]

**Pull request merged**